MedallionShell.Tests/Streams/AsyncEnumerableAdapter.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Medallion.Shell.Tests.Streams;
+
+public class AsyncEnumerableAdapter : IAsyncEnumerable<string>
+{
+    private readonly IEnumerable<string> strings;
+
+    public AsyncEnumerableAdapter(IEnumerable<string> strings)
+    {
+        this.strings = strings;
+    }
+
+    public IAsyncEnumerator<string> GetAsyncEnumerator(CancellationToken cancellationToken = default) =>
+        // this does not allow consuming the same IEnumerable twice
```

Comment:
@madelson  `TestConsumeTwice` started failing after making this change. Explanation from ChatGPT
> In the `AsyncEnumerableAdapter` class, the `GetAsyncEnumerator()` method creates a new `AsyncEnumeratorAdapter` instance that wraps the original `IEnumerator<string>`. This means that it's directly using the original enumerator from the `IEnumerable<string>`, which can only be enumerated once. If you try to get the async enumerator twice, it will fail because the underlying enumerator has already been exhausted.
> 
> On the other hand, the `AsAsyncEnumerable()` extension method creates a new enumerator each time it's called. It does this by iterating over the `IEnumerable<T>` items in a `foreach` loop. This means you can call `GetAsyncEnumerator()` multiple times without an issue because each call creates a new enumerator.
> 
> So, if your test expects an `InvalidOperationException` when calling `GetAsyncEnumerator()` twice, it will fail when using `AsAsyncEnumerable()` because this method allows multiple enumerations. If you want to preserve the single-use behavior, you should stick with the `AsyncEnumerableAdapter` class. If you want to allow multiple enumerations, then `AsAsyncEnumerable()` is the way to go. It all depends on the specific requirements of your code.

For posterity, I added this comment.